gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/LocationMapper.java

```diff
@@ -0,0 +1,128 @@
+        var identifier = getIdentifier(id);
+
+        if (location.getLocatedEntity() != null && location.getLocatedEntity().getLocatedPlace() != null) {
+            var locatedPlace = location.getLocatedEntity().getLocatedPlace();
+            var name = getName(locatedPlace);
+            var telecom = getTelecom(locatedPlace);
+            var address = getAddress(locatedPlace);
+
+            return createLocation(id, identifier, name, telecom, address);
+        }
+
+        return createLocation(id, identifier, null, null, null);
+    }
+
+    private Identifier getIdentifier(String id) {
+        Identifier identifier = new Identifier()
+                .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL
+                .setValue(id);
+        return identifier;
+    }
+
+    private String getName(RCMRMT030101UK04Place locatedPlace) {
+        var name = locatedPlace.getName();
+        return name != null ? name : UNKNOWN_NAME;
+    }
+
+    private ContactPoint getTelecom(RCMRMT030101UK04Place locatedPlace) {
+        var telecom = locatedPlace.getTelecom().stream().findFirst();
+        if (telecom.isPresent()) {
+            var contactPoint = new ContactPoint();
+            return contactPoint
+                    .setValue(getValue(telecom.get()))
+                    .setSystem(ContactPoint.ContactPointSystem.PHONE)
+                    .setUse(ContactPoint.ContactPointUse.WORK)
+                    .setRank(TELECOM_RANK);
+        }
+
+        return null;
+    }
+
+    private String getValue(TEL telecom) {
+        var value = telecom.getValue();
+        if (value != null) {
+            return isWorkPlaceValue(telecom.getUse()) ? stripTelPrefix(value) : value;
+        }
+
+        return null;
+    }
+
+    private boolean isWorkPlaceValue(List<CsTelecommunicationAddressUse> use) {
+        if (!use.isEmpty()) {
+            return use.stream().findFirst().get().value().equals(WORK_PLACE);
+        }
+
+        return false;
+    }
+
+    private String stripTelPrefix(String value) {
+        return value.substring(TEL_PREFIX_INT); // strips the 'tel:' prefix on phone numbers
+    }
+
+    private Address getAddress(RCMRMT030101UK04Place locatedPlace) {
+        var address = locatedPlace.getAddr();
+        if (isValidAddress(address)) {
+            var mappedAddress = new Address();
+            if (address.getStreetAddressLine() != null) {
+                address.getStreetAddressLine()
+                        .forEach(addressLine -> mappedAddress.addLine(addressLine));
+            }
+            if (StringUtils.isNotEmpty(address.getPostalCode())) {
+                mappedAddress.setPostalCode(address.getPostalCode());
+            }
+
+            return mappedAddress;
+        }
+        return null;
+    }
+
+    private boolean isValidAddress(AD address) {
+        if (address != null && address.getUse() != null) {
+            var use = address.getUse().stream().findFirst();
```

Comment:
Could you try to change this method to filter by value as well, similarly to `isWorkPlaceValue()`?

Reply:
Added anyMatch filter to both mehtods you mentioned, thanks! 😄